### PR TITLE
Stack upgrades

### LINKED ISSUE #3299: Stack Upgrades
# Angular
- fix comment line lengths
# React
- Angular has `analyticsString` and there is no harm in adding it to React, so we can easily add that
- Remove empty default props from SprkStack and SprkStackItem
- Fix line lengths

### PR DESCRIPTION
**Pull request by bobwise**
# Angular
- fix comment line lengths
# React
- Adds 'analyticsString' to React Stack
- Remove empty default props from SprkStack and SprkStackItem
- Fix line lengths

### Associated Issue
Fixes #3299 

### Documentation
 - [x] Update Spark Docs

### Code
 - [x] Build Component in Angular
 - [x] Build Component in React
 - [x] Unit Testing in Angular with `npm run test` in `angular/` (100% coverage, 100% passing)
 - [x] Unit Testing in React with `npm run test` in `react/` (100% coverage, 100% passing)

### Browser Testing (current version and 1 prior)
  - [x] Google Chrome
  - [x] Google Chrome (Mobile)
  - [x] Mozilla Firefox
  - [x] Mozilla Firefox (Mobile)
  - [x] Microsoft Internet Explorer 11 (only this specific version of IE)
  - [x] Microsoft Edge
  - [x] Apple Safari
  - [x] Apple Safari (Mobile)
  
### Accessibility Testing
  - [x] Axe browser extension

### CODE REVIEW

File: react/src/objects/stack/SprkStack.stories.js [hunk 1]
```diff
@@ -173,64 +169,130 @@ stackSplitLayoutThreeTenths.story = {
 export const stackSplitLayoutMixed = () => (
   <div>
     <SprkStack splitAt="tiny">
-      <SprkStackItem additionalClasses="sprk-o-Stack__item--fourth@xs sprk-u-AbsoluteCenter">
+      <SprkStackItem
+        additionalClasses="
+        sprk-o-Stack__item--fourth@xs
+        sprk-u-AbsoluteCenter"
```
Comment (guern1kn) [inline react/src/objects/stack/SprkStack.stories.js:175]: Need an extra indent here

File: react/src/objects/stack/SprkStack.stories.js [hunk 2]
```diff
@@ -173,64 +169,130 @@ stackSplitLayoutThreeTenths.story = {
...
+      >
         <p className="sprk-b-TypeBodyOne">fourth</p>
       </SprkStackItem>
-      <SprkStackItem additionalClasses="sprk-o-Stack__item--half@xs sprk-u-AbsoluteCenter">
+
```
Comment (guern1kn) [inline react/src/objects/stack/SprkStack.stories.js:179]: Extra line?

File: react/src/objects/stack/SprkStack.stories.js [hunk 3]
```diff
@@ -173,64 +169,130 @@ stackSplitLayoutThreeTenths.story = {
...
+            additionalClasses="
+              sprk-o-Stack__item--flex@xs
+              sprk-u-AbsoluteCenter"
+          >
             <p className="sprk-b-TypeBodyOne">Nested Item (flex)</p>
           </SprkStackItem>
         </SprkStack>
       </SprkStackItem>
-      <SprkStackItem additionalClasses="sprk-o-Stack__item--half@xs sprk-u-AbsoluteCenter">
+
```
Comment (guern1kn) [inline react/src/objects/stack/SprkStack.stories.js:274]: Extra line?

File: react/src/objects/stack/SprkStack.test.js [hunk 4]
```diff
@@ -9,164 +9,215 @@ Enzyme.configure({ adapter: new Adapter() });
...
+      const wrapper = shallow(<SprkStack itemSpacing="large" />);
+      expect(wrapper.find('div.sprk-o-Stack.sprk-o-Stack--large').length).toBe(
+        1,
+      );
+    },
+  );
+
+  it(
+    'should set the huge spacing class in addition to the base class when' +
+      'itemSpacing is "huge"',
```
Comment (guern1kn) [inline react/src/objects/stack/SprkStack.test.js:61]: space before itemSpacing

File: react/src/objects/stack/SprkStack.test.js [hunk 5]
```diff
@@ -9,164 +9,215 @@ Enzyme.configure({ adapter: new Adapter() });
...
+      const wrapper = shallow(<SprkStack itemSpacing="huge" />);
+      expect(wrapper.find('div.sprk-o-Stack.sprk-o-Stack--huge').length).toBe(
+        1,
+      );
+    },
+  );
+
+  it(
+    'should set the misc-a spacing class in addition to the base class when' +
+      'itemSpacing is "misc-a"',
```
Comment (guern1kn) [inline react/src/objects/stack/SprkStack.test.js:72]: space before itemSpacing

File: react/src/objects/stack/SprkStack.test.js [hunk 6]
```diff
@@ -9,164 +9,215 @@ Enzyme.configure({ adapter: new Adapter() });
...
+      const wrapper = shallow(<SprkStack itemSpacing="misc-a" />);
+      expect(wrapper.find('div.sprk-o-Stack.sprk-o-Stack--misc-a').length).toBe(
+        1,
+      );
+    },
+  );
+
+  it(
+    'should set the misc-a spacing class in addition to the base class when' +
+      'itemSpacing is "miscA"',
```
Comment (guern1kn) [inline react/src/objects/stack/SprkStack.test.js:83]: space before itemSpacing

File: react/src/objects/stack/SprkStack.test.js [hunk 7]
```diff
@@ -9,164 +9,215 @@ Enzyme.configure({ adapter: new Adapter() });
...
+      const wrapper = shallow(<SprkStack itemSpacing="miscA" />);
+      expect(wrapper.find('div.sprk-o-Stack.sprk-o-Stack--misc-a').length).toBe(
+        1,
+      );
+    },
+  );
+
+  it(
+    'should set the misc-b spacing class in addition to the base class when' +
+      'itemSpacing is "misc-b"',
```
Comment (guern1kn) [inline react/src/objects/stack/SprkStack.test.js:94]: space here

File: react/src/objects/stack/SprkStack.test.js [hunk 8]
```diff
@@ -9,164 +9,215 @@ Enzyme.configure({ adapter: new Adapter() });
...
+      const wrapper = shallow(<SprkStack itemSpacing="misc-b" />);
+      expect(wrapper.find('div.sprk-o-Stack.sprk-o-Stack--misc-b').length).toBe(
+        1,
+      );
+    },
+  );
+
+  it(
+    'should set the misc-b spacing class in addition to the base class when' +
+      'itemSpacing is "miscB"',
```
Comment (guern1kn) [inline react/src/objects/stack/SprkStack.test.js:105]: space

File: react/src/objects/stack/SprkStack.test.js [hunk 9]
```diff
@@ -9,164 +9,215 @@ Enzyme.configure({ adapter: new Adapter() });
 describe('SprkStack:', () => {
```
Comment (guern1kn) [inline react/src/objects/stack/SprkStack.test.js:9]: Need to add a test for `analyticsString` prop

File: react/src/objects/stack/SprkStack.stories.js [hunk 10]
```diff
@@ -110,11 +106,11 @@ stackSplitFourCol.story = {
 
 export const stackSplitFiveCol = () => (
   <SprkStack splitAt="tiny">
-    <SprkStackItem additionalClasses="sprk-o-Stack__item--fifth@xs"></SprkStackItem>
-    <SprkStackItem additionalClasses="sprk-o-Stack__item--fifth@xs"></SprkStackItem>
-    <SprkStackItem additionalClasses="sprk-o-Stack__item--fifth@xs"></SprkStackItem>
-    <SprkStackItem additionalClasses="sprk-o-Stack__item--fifth@xs"></SprkStackItem>
-    <SprkStackItem additionalClasses="sprk-o-Stack__item--fifth@xs"></SprkStackItem>
+    <SprkStackItem additionalClasses="sprk-o-Stack__item--fifth@xs" />
```
Comment (afebbraro) [inline react/src/objects/stack/SprkStack.stories.js:109]: @bobwise do you think removing the closing tag makes it more difficult for users of Spark?

Comment (bobwise) [reply react/src/objects/stack/SprkStack.stories.js:109]: @afebbraro This is the only way I could get it under 80 characters. If I put it on a new line Prettier undoes it automatically. I think this is definitely more difficult for Spark users.

Comment (afebbraro) [reply react/src/objects/stack/SprkStack.stories.js:109]: @bobwise ah okay. In the case where linting hurts usability for our consumers, lets disable it for that line or lines.

**Pull request merged**